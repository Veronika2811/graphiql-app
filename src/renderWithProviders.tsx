import { PropsWithChildren } from 'react';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import { render, type RenderOptions } from '@testing-library/react';
import { SnackbarProvider } from 'context/snackbar-provider';
import { LocaleProvider } from 'internationalization/LocaleProvider';
import { MuiProvider } from 'layouts/mui';
import { AppStore, setupStore } from 'store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  store?: AppStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  { store = setupStore(), ...renderOptions }: ExtendedRenderOptions = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren): JSX.Element => (
    <MuiProvider>
      <SnackbarProvider>
        <HashRouter>
          <Provider store={store}>
            <LocaleProvider>{children}</LocaleProvider>
          </Provider>
        </HashRouter>
      </SnackbarProvider>
    </MuiProvider>
  );
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export default renderWithProviders;
