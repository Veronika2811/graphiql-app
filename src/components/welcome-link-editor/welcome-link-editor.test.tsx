import { renderWithProviders } from '../../renderWithProviders';

import { WelcomeLinkEditor } from '.';

describe('WelcomeLinkEditor component', () => {
  it('should renders correctly WelcomeLinkEditor component', () => {
    const container = renderWithProviders(<WelcomeLinkEditor />);
    expect(container).toMatchSnapshot();
  });
});
