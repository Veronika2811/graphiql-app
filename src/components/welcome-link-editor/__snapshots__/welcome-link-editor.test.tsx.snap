// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WelcomeLinkEditor component > should renders correctly WelcomeLinkEditor component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignRight css-46qt88-MuiTypography-root"
      >
        <a
          aria-label="link-editor"
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-m2vvas-MuiTypography-root-MuiLink-root"
          href="#/editor"
        >
          Main Page
        </a>
      </h2>
    </div>
  </body>,
  "container": <div>
    <h2
      class="MuiTypography-root MuiTypography-h3 MuiTypography-alignRight css-46qt88-MuiTypography-root"
    >
      <a
        aria-label="link-editor"
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-m2vvas-MuiTypography-root-MuiLink-root"
        href="#/editor"
      >
        Main Page
      </a>
    </h2>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
